File: coding_skill_py/23_watchdog_pattern.py
"""Python Watchdog - time 모듈"""
import time

class Watchdog:
    def __init__(self, timeout: int):
        self.timeout = timeout
        self.kick()
    
    def kick(self):
        self.last_kick = time.time()
    
    def check(self) -> bool:
        elapsed = time.time() - self.last_kick
        return elapsed < self.timeout

def main():
    print("=== Python Watchdog ===")
    wd = Watchdog(3)
    print(f"Status: {'OK' if wd.check() else 'Timeout'}")

if __name__ == "__main__":
    main()

File: coding_skill_cpp/23_watchdog_pattern.cpp
/* C++ Watchdog - chrono */
#include <iostream>
#include <chrono>
using namespace std;
using namespace chrono;

class Watchdog {
    time_point<steady_clock> lastKick;
    int timeoutSec;
public:
    Watchdog(int timeout) : timeoutSec(timeout) { kick(); }
    void kick() { lastKick = steady_clock::now(); }
    bool check() {
        auto elapsed = duration_cast<seconds>(steady_clock::now() - lastKick).count();
        return elapsed < timeoutSec;
    }
};

int main() {
    cout << "=== C++ Watchdog ===" << endl;
    Watchdog wd(3);
    cout << "Status: " << (wd.check() ? "OK" : "Timeout") << endl;
    return 0;
}
